dataframe/time_series.go

```diff
@@ -0,0 +1,369 @@
+package dataframe
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strconv"
+	"time"
+)
+
+// TimeSeriesType represents the type of time series the schema can be treated as (if any).
+type TimeSeriesType int
+
+// TODO: Create and link to Grafana documentation on Long vs Wide
+const (
+	// TimeSeriesTypeNot means this dataframe is not a valid time series. This means it lacks at least
+	// one of a time Field and a number Field.
+	TimeSeriesTypeNot TimeSeriesType = iota
+	// TimeSeriesTypeLong means this dataframe can be treated as a "Long" time series.
+	//
+	// A Long series has one or more string Fields, disregards Labels on Fields, and generally
+	// repeated time values in the time index.
+	TimeSeriesTypeLong
+	// TimeSeriesTypeLong means this dataframe can be treated as a "Wide" time series.
+	//
+	// A Wide series has no string fields, should not have repeated time values, and generally
+	// uses labels.
+	TimeSeriesTypeWide
+)
+
+func (t TimeSeriesType) String() string {
+	switch t {
+	case TimeSeriesTypeLong:
+		return "long"
+	case TimeSeriesTypeWide:
+		return "wide"
+	}
+	return "not"
+}
+
+// TimeSeriesSchema returns the TimeSeriesSchema of the frame. The TimeSeriesSchema's Type
+// value will be TimeSeriesNot if it is not a time series.
+func (f *Frame) TimeSeriesSchema() (tsSchema TimeSeriesSchema) {
```

Comment:
named returns? Not sure how I feel about that. Definitely not something we use in grafana code base unless needed. Ex `recover()`. But let's postpone that topic :)

Reply:
There are a lot of returns in the function body, and only one thing is returned. So it makes the body of function easier to read.

Reply:
Agree. I'm just not used to it :)

Let's keep it like this.